apis/projectcontour/v1/register.go

```diff
@@ -33,12 +33,30 @@ func Resource(resource string) schema.GroupResource {
 	return SchemeGroupVersion.WithResource(resource).GroupResource()
 }
 
-func AddKnownTypes(scheme *runtime.Scheme) {
+// AddKnownTypes is exported for backwards compatibility with third
+// parties who depend on this symbol, but all new code should use
+// AddToScheme.
+func AddKnownTypes(scheme *runtime.Scheme) error {
 	scheme.AddKnownTypes(SchemeGroupVersion,
 		&HTTPProxy{},
 		&HTTPProxyList{},
 		&TLSCertificateDelegation{},
 		&TLSCertificateDelegationList{},
 	)
 	metav1.AddToGroupVersion(scheme, SchemeGroupVersion)
+	return nil
 }
+
+// The following declarations are kubebuilder-compatible and will be expected
+// by third parties who import the Contour API types.
+
+var (
+	// GroupVersion is group version used to register these objects
+	GroupVersion = schema.GroupVersion{Group: GroupName, Version: "v1"}
```

Comment:
We already have this field above with a different name, maybe merge them? (https://github.com/projectcontour/contour/blob/751a0ebc939d14ce16bd31469d86082896c3cb09/apis/projectcontour/v1/register.go#L27)

Reply:
Yeh, I just got paranoid about breaking someone that imports it? I added a comment that is is for compatibility, but we can remove it if we think that's a low risk change.

Reply:
We need to keep both names - that's the whole point of this exercise, sadly. I think James' solution for now is the best we're going to get.

Reply:
'kk, leaving this for now.